### DISCUSSION TITLE
editing yields "NetworkError when attempting to fetch resource"

### DISCUSSION THREAD

**Discussion by UltraBlackLinux** [dani-garcia/vaultwarden#2711] [Q&A] [answered]
### Subject of the issue
Whenever I try to edit an item or folder (moving, renaming, whatever) I get [this error](https://github.com/dani-garcia/vaultwarden/files/9438128/-1661609993223.log). The only thing that works is creating an entry

### Deployment environment

* vaultwarden version: latest

* Install method: docker image

* Clients used:  Web vault, desktop

* Reverse proxy and version:  nginx version: nginx/1.18.0 (Ubuntu)

* MySQL/MariaDB or PostgreSQL version:  none (I didn't change anything at least)

* Other relevant details: I'm using firewalld and just imported my whole KeePassXC config via the KeePassX import option.

### Steps to reproduce

I started the vaultwarden container as advised in the wiki

### Expected behaviour

Editing anything should work

### Actual behaviour

Everytime I try to edit or create anything I get the error mentioned above. Creating new passwords works

### Troubleshooting data

[-1661609993223.log](https://github.com/dani-garcia/vaultwarden/files/9438128/-1661609993223.log)

**Comment by BlackDex** [comment]
Could you provide logs from Vaultwarden?? The logs attached aren't useful, these look like client logs, and that is not something on the Vaultwarden side we can fix.

Also check the nginx logs.

**Comment by UltraBlackLinux** [comment]
> Could you provide logs from Vaultwarden?? The logs attached aren't useful, these look like client logs, and that is not something on the Vaultwarden side we can fix.
> 
> Also check the nginx logs.

I thought so, but I couldn't get my hands on any useful logs. The only ones I'm seeing are login logs

EDIT: Almost forgot the nginx logs: it doesn't look like any recent errors exist in there, and the access log shows my deletion attempt

**Comment by BlackDex** [comment]
Could you then try to provide the support string from the admin diagnostics page? (`/admin`)

**Comment by UltraBlackLinux** [comment]
I can't get the admin page to work. Currently vaultwarden is hosted at domain/vaultwarden/ via proxy_pass - This works perfectly, but the admin panel redirects me to domain/admin which is not the vaultwarden location

Also css seems broken on the admin page. Proxy pass is just not great

EDIT: Managed to access it via the ip directly. This is apparently the default which works.

**Comment by UltraBlackLinux** [comment]
Okay! Got it. Took some time to deal with various issues, but here it is:
### Your environment (Generated via diagnostics page)
* Vaultwarden version: v1.25.2
* Web-vault version: v2022.6.2
* Running within Docker: true (Base: Debian)
* Environment settings overridden: true
* Uses a reverse proxy: true
* IP Header check: true (X-Real-IP)
* Internet access: true
* Internet access via a proxy: false
* DNS Check: true
* Time Check: true
* Domain Configuration Check: false
* HTTPS Check: false
* Database type: SQLite
* Database version: 3.35.4
* Clients used: 
* Reverse proxy and version: 
* Other relevant information: 

### Config (Generated via diagnostics page)
<details><summary>Show Running Config</summary>

**Environment settings which are overridden:** ADMIN_TOKEN


```json
{
  "_duo_akey": null,
  "_enable_duo": false,
  "_enable_email_2fa": false,
  "_enable_smtp": true,
  "_enable_yubico": true,
  "_icon_service_csp": "",
  "_icon_service_url": "",
  "_ip_header_enabled": true,
  "admin_ratelimit_max_burst": 3,
  "admin_ratelimit_seconds": 300,
  "admin_token": "***",
  "allowed_iframe_ancestors": "",
  "attachments_folder": "data/attachments",
  "authenticator_disable_time_drift": false,
  "data_folder": "data",
  "database_conn_init": "",
  "database_max_conns": 10,
  "database_timeout": 30,
  "database_url": "****/**.*******",
  "db_connection_retries": 15,
  "disable_2fa_remember": false,
  "disable_admin_token": false,
  "disable_icon_download": false,
  "domain": "****://******.************.**/***********",
  "domain_origin": "****://******.************.**",
  "domain_path": "/***********",
  "domain_set": true,
  "duo_host": null,
  "duo_ikey": null,
  "duo_skey": null,
  "email_attempts_limit": 3,
  "email_expiration_time": 600,
  "email_token_size": 6,
  "emergency_access_allowed": true,
  "emergency_notification_reminder_schedule": "0 5 * * * *",
  "emergency_request_timeout_schedule": "0 5 * * * *",
  "enable_db_wal": true,
  "extended_logging": true,
  "helo_name": null,
  "hibp_api_key": null,
  "icon_blacklist_non_global_ips": true,
  "icon_blacklist_regex": null,
  "icon_cache_folder": "data/icon_cache",
  "icon_cache_negttl": 259200,
  "icon_cache_ttl": 2592000,
  "icon_download_timeout": 10,
  "icon_redirect_code": 302,
  "icon_service": "internal",
  "incomplete_2fa_schedule": "30 * * * * *",
  "incomplete_2fa_time_limit": 3,
  "invitation_org_name": "Vaultwarden",
  "invitations_allowed": true,
  "ip_header": "X-Real-IP",
  "job_poll_interval_ms": 30000,
  "log_file": "/data/vaultwarden.log",
  "log_level": "Info",
  "log_timestamp_format": "%Y-%m-%d %H:%M:%S.%3f",
  "login_ratelimit_max_burst": 10,
  "login_ratelimit_seconds": 60,
  "org_attachment_limit": null,
  "org_creation_users": "",
  "password_hints_allowed": true,
  "password_iterations": 100000,
  "reload_templates": false,
  "require_device_email": false,
  "rsa_key_filename": "data/rsa_key",
  "send_purge_schedule": "0 5 * * * *",
  "sends_allowed": true,
  "sends_folder": "data/sends",
  "show_password_hint": false,
  "signups_allowed": false,
  "signups_domains_whitelist": "",
  "signups_verify": false,
  "signups_verify_resend_limit": 6,
  "signups_verify_resend_time": 3600,
  "smtp_accept_invalid_certs": false,
  "smtp_accept_invalid_hostnames": false,
  "smtp_auth_mechanism": null,
  "smtp_debug": false,
  "smtp_explicit_tls": null,
  "smtp_from": "",
  "smtp_from_name": "Vaultwarden",
  "smtp_host": null,
  "smtp_password": null,
  "smtp_port": 587,
  "smtp_security": "starttls",
  "smtp_ssl": null,
  "smtp_timeout": 15,
  "smtp_username": null,
  "templates_folder": "data/templates",
  "tmp_folder": "data/tmp",
  "trash_auto_delete_days": null,
  "trash_purge_schedule": "0 5 0 * * *",
  "use_syslog": false,
  "user_attachment_limit": null,
  "web_vault_enabled": true,
  "web_vault_folder": "web-vault/",
  "websocket_address": "0.0.0.0",
  "websocket_enabled": false,
  "websocket_port": 3012,
  "yubico_client_id": null,
  "yubico_secret_key": null,
  "yubico_server": null
}
```
</details>

EDIT: Some client side stuff that I just noticed:
```
Connection to xyz failed: Error during WebSocket handshake: Unexpected response code: 400

Failed to start the connection: Error: There was an error with the transport
```

**Comment by UltraBlackLinux** [comment]
Yikes. It seems to be proxy related; Just entering my server ip and directory allows me to delete stuff as expected

I just noticed that I did get some kind of logs from my previous attempt now:
```
[2022-08-27 15:22:36.817][_][ERROR] No matching routes for GET / text/html.
[2022-08-27 15:22:36.818][_][WARN] No 404 catcher registered. Using Rocket default.
[2022-08-27 15:28:59.691][_][ERROR] No matching routes for POST //identity/connect/token application/x-www-form-urlencoded.
[2022-08-27 15:28:59.691][_][WARN] No 404 catcher registered. Using Rocket default.
```
I can't log in from the web anymore

EDIT; HUH? My config is causing that? How is that possible

**Comment by BlackDex** [comment]
It looks like you configured your reverse proxy incorrectly.
Because you are using a sub-path it needs some different configuration they just a default proxy pass.
I suggest to double check the wiki regarding this, and see if your configuration matches.
https://github.com/dani-garcia/vaultwarden/wiki/Proxy-examples and then check my example `Nginx with sub-path`.

The reason the admin interface isn't working, is because either the reverse proxy is faulty, or your `DOMAIN` config variable is incorrect. Once you have fixed your reverse proxy, the admin interface will probably work, and then the web-vault also will work just fine.

**Comment by UltraBlackLinux** [comment]
Still no luck, just 404s. I can't even get to the web interface now

**Reply by BlackDex** [reply]
Can you post your nginx config?

**Reply by UltraBlackLinux** [reply]
server section:
```nginx
server_name <domain>;
listen 443 ssl;

location /vw/ {
      proxy_http_version 1.1;
      proxy_set_header "Connection" "";

      proxy_set_header Host $host;
      proxy_set_header X-Real-IP $remote_addr;
      proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
      proxy_set_header X-Forwarded-Proto $scheme;

      proxy_pass http://127.0.0.1:8002/;
    }

    location /vw/notifications/hub/negotiate {
      proxy_http_version 1.1;
      proxy_set_header "Connection" "";

      proxy_set_header Host $host;
      proxy_set_header X-Real-IP $remote_addr;
      proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
      proxy_set_header X-Forwarded-Proto $scheme;

      proxy_pass http://127.0.0.1:8002/;
    }

    location /vw/notifications/hub {
      proxy_http_version 1.1;
      proxy_set_header Upgrade $http_upgrade;
      proxy_set_header Connection "upgrade";

      proxy_set_header Host $host;
      proxy_set_header X-Real-IP $remote_addr;
      proxy_set_header Forwarded $remote_addr;
      proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
      proxy_set_header X-Forwarded-Proto $scheme;

      proxy_pass http://127.0.0.1:8002/;
    }
```

I tried to downgrade this as much as possible from the big config you sent me. I tried even that, but nothing works. Where even do I have to set the DOMAIN variable? I currently have it set via docker run ... -e DOMAIN=foobar

**Reply by BlackDex** [reply]
Ok, looking at your `DOMAIN` you configured, it looks like the path is different from `/vw/`, if they do not match, that will break.
Secondly, it's a small thing, but remove the trailing `/` at all the `proxy_pass` endpoints.
Third, the `/vw/notifications/hub` seems to directed to the exact same port as the rest, that is probably a miss configuration.
That should be direct to the port-forward you configured for `3012`.

**Reply by UltraBlackLinux** [reply]
1. I thought the DOMAIN variable would override the config option. All this time I was using the default config since every time I activated mine I just got a 404
2. I'm actually gonna die I think. That brought back my ui
3. The docker container only has one port which does everything.

Edit: Same error as way in the beginning.

**Reply by UltraBlackLinux** [reply]
Why does all of this have to be so complicated? Take a look at Gitea for example: 
```nginx
location /git/ {
	proxy_pass http://localhost:3000/;
	client_max_body_size 500M;
        proxy_set_header Host $host;
        proxy_set_header X-Real-IP $remote_addr;
        proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
        proxy_set_header X-Forwarded-Proto $scheme;
	}
```
This is everything that it needs, and it has similar client-server interaction like bitwarden

**Reply by BlackDex** [reply]
Well, you can disable the websockets, and then have just one entry, like gitea, but then websockets will not work.
The env variables are overridden by the `config.json` values, so it's the other way around.

**Reply by BlackDex** [reply]
> 1. I thought the DOMAIN variable would override the config option. All this time I was using the default config since every time I activated mine I just got a 404
> 
>     2. I'm actually gonna die I think. That brought back my ui
> 
>     3. The docker container only has one port which does everything.
> 
> 
> Edit: Same error as way in the beginning.

1. If there is a `config.json` it will overrule the ENV variables.
2. Please don't
3. The docker container has two exposes, one for the web-vault it self, and one for the web-sockets.
    The reason for this is that Rocket (The web-server library) doesn't support web-sockets it self (yet). For this we need to use a different library and thus a different port.

**Comment by UltraBlackLinux** [comment, marked as answer]
I gave up

**Reply by BlackDex** [reply]
Try to rename the config.json to something else. And restart. And make sure the environment variables are correct.

**Reply by UltraBlackLinux** [reply]
tried that a while ago. I was advised to try caddy instead of nginx since it's much easier to use. Once I have that set up I will give this another try. I've wasted many hours today
